src/math/TransformBoundedField.cc

```diff
@@ -0,0 +1,172 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include <memory>
+#include <string>
+
+#include "ndarray/eigen.h"
+#include "astshim.h"
+#include "lsst/afw/math/TransformBoundedField.h"
+#include "lsst/afw/table/io/InputArchive.h"
+#include "lsst/afw/table/io/OutputArchive.h"
+#include "lsst/afw/table/io/CatalogVector.h"
+#include "lsst/afw/table/aggregates.h"
+
+namespace lsst {
+namespace afw {
+namespace math {
+
+TransformBoundedField::TransformBoundedField(
+        geom::Box2I const& bbox,
+        geom::Transform<geom::Point2Endpoint, geom::GenericEndpoint> const& transform)
+        : BoundedField(bbox), _transform(transform) {
+    if (transform.getToEndpoint().getNAxes() != 1) {
+        throw LSST_EXCEPT(pex::exceptions::InvalidParameterError,
+                          "To EndPoint of transform must have 1 axis but has " +
+                                  std::to_string(transform.getToEndpoint().getNAxes()));
+    }
+}
+
+double TransformBoundedField::evaluate(geom::Point2D const& position) const {
+    return _transform.applyForward(position)[0];
+}
+
+ndarray::Array<double, 1, 1> TransformBoundedField::evaluate(ndarray::Array<double const, 1> const& x,
+                                                             ndarray::Array<double const, 1> const& y) const {
+    if (x.getSize<0>() != y.getSize<0>()) {
+        throw LSST_EXCEPT(pex::exceptions::InvalidParameterError,
+                          "x length " + std::to_string(x.getSize<0>()) + "!= y length " +
+                                  std::to_string(x.getSize<0>()));
+    }
+
+    // TODO if Mapping.applyForward gains support for x, y then use that instead of copying data
+    int const nPoints = x.getSize<0>();
+    ndarray::Array<double, 2, 2> xy = ndarray::allocate(ndarray::makeVector(2, nPoints));
+    for (int col = 0; col < nPoints; ++col) {
+        xy[0][col] = x[col];
+        xy[1][col] = y[col];
+    }
+
+    auto res2D = _transform.getFrameSet()->applyForward(xy);
+
+    // res2D has shape 1 x N; return a 1-D view with the extra dimension stripped
+    auto resShape = ndarray::makeVector(nPoints);
+    auto resStrides = ndarray::makeVector(1);
+    return ndarray::external(res2D.getData(), resShape, resStrides, res2D);
+}
+
+// ------------------ persistence ---------------------------------------------------------------------------
+
+namespace {
+
+struct PersistenceHelper {
+    table::Schema schema;
+    table::Key<int> orderX;
+    table::PointKey<int> bboxMin;
+    table::PointKey<int> bboxMax;
+    table::Key<std::string> frameSet;
+
+    PersistenceHelper()
+            : schema(),
```

Comment:
`orderX` is not supposed to exist; it was a holdover from `ChebyshevBoundedField`. Thanks for catching that.